cli/tools/test/mod.rs

```diff
@@ -466,6 +491,10 @@ pub async fn test_specifier(
   worker.dispatch_beforeunload_event(located_script_name!())?;
   worker.dispatch_unload_event(located_script_name!())?;
 
+  // Ensure the worker has settled so we can catch any remaining unhandled rejections. We don't
+  // want to wait forever here.
+  worker.run_up_to_duration(Duration::from_millis(0)).await?;
```

Comment:
This is a bit concerning to me, why unhandled rejections are not caught by the previous line that dispatch the unload event? Does that mean that we need to update all callsites in the whole codebase to now catch these rejections properly?

Reply:
I am currently separating out the functionality of our "regular" event-loop driving code (ie: poll_event_loop) and our "immediate" event-loop driving code (call_and_await, resolve_promise). Unhandled promise rejections are a concern of the "regular" mode of operation, as any number of async tasks may be raised by immediate calls.

Most of this code works only by luck, and only for rejections handled before the function returns.